discovery-provider/src/tasks/tracks.py

```diff
@@ -204,8 +204,50 @@ def wrapper(*args, **kargs):
     return wrapper
 
 
+@time_method
+def add_old_style_route(session, track_record, track_metadata):
+    """Temporary method to add the old style routes to the track_routes db while we
+    transition the client to use the new routing API.
+    """
+    try:
+        # Use create_track_route_id() since the regex replace is slightly different
+        new_track_slug_title = helpers.create_track_route_id(
+            track_metadata["title"], ""
+        )
+        # Remove "/"
+        new_track_slug_title = new_track_slug_title[1:]
+        # Append ID
+        new_track_slug_title = f"{new_track_slug_title}-{track_record.track_id}"
+
+        # Add the new track route
+        new_track_route = TrackRoute()
+        new_track_route.slug = new_track_slug_title
+        new_track_route.title_slug = new_track_slug_title
+        new_track_route.collision_id = 0
+        new_track_route.owner_id = track_record.owner_id
+        new_track_route.track_id = track_record.track_id
+        new_track_route.is_current = (
+            False  # This route is meant to be a fallback, not the main route
+        )
+        new_track_route.blockhash = track_record.blockhash
+        new_track_route.blocknumber = track_record.blocknumber
+        new_track_route.txhash = track_record.txhash
+        session.add(new_track_route)
+
+        # Commit this so it gets in before the new route creation
+        session.commit()
```

Comment:
just wondering if there could be any unintended consequences of committing here if we haven't committed other changes earlier. i think fine b/c this is at the start of the indexing flow. maybe worth a comment inline where it's called to the future editor knows that it does commit here

Reply:
Note that in the "new-style" routes method I also commit. I added some comments where these get called.

also cc: @hareeshnagaraj

Reply:
@rickyrombo @raymondjacobson is the reason we need to commit because inside a postgres tx we can't read data uncommitted?

Reply:
just to make sure i understand. since we now call session.commit here, one set of db records is written with user + some track changes. then the second commit writes reset of the indexing changes like ursm, socials etc.

this changes the way we do block level commits atomically for all records. does this affect rollbacks?

Reply:
Sorry just seeing this now. It does disrupt the unit of work model of each block. We could keep track of the unwritten routes separately in memory for each transaction, which could avoid this commit. I can add that in a PR next week?

Reply:
I haven't tested rollbacks - any ideas/suggestions there? - but I imagine this would disrupt that.

Reply:
I don't think this actually breaks anything because these commits all fire before we touch the track record, etc. The in-memory solution seems fine, but I also don't think it's necessary.

If for some reason we hit an exception after writing to the track routes table, we would effectively have considered indexing that block a failure and would retry it. Even if the routes table changes were not reverted, the next time the indexer tries to re-index that block, it would try to re update the routes and because they're (IIUC) idempotent, we would continue to make progress.

Worst-case, we have halted indexing entirely and we would skip over the block (after consensus) per @jowlee 's work and all discovery nodes would be consistent. Because the routes table model supports backlinks in various capacity, that would be fine.

This definitely all changes the "unit of work" model, but I don't think for the worse